apps/files/js/breadcrumb.js

```diff
@@ -163,19 +189,20 @@
 				if(menuPart.dir) {
 					$menuItem = $('<li class="crumblist"><a><span class="icon-folder"></span><span></span></a></li>');
 					$menuItem.data('dir', menuPart.dir);
-					$menuItem.find('a').attr('href', this.getCrumbUrl(part, j));
+					$menuItem.find('a').attr('href', this.getCrumbUrl(parts[Math.max(this.activeItemIndex, 0)], j));
 					$menuItem.find('span:eq(1)').text(menuPart.name);
 					this.$menu.children('ul').append($menuItem);
 					if (this.onClick) {
 						$menuItem.on('click', this.onClick);
 					}
 				}
 			}
+
 			_.each(this._detailViews, function(view) {
 				view.render({
 					dirInfo: this.dirInfo
 				});
-				$crumb.append(view.$el);
+				this.breadcrumbs[this.activeItemIndex + 2].append(view.$el);
```

Comment:
While testing, there was an error when `this.dir === '/'. This helped:

```suggestion
				if (this.breadcrumbs.length > 2) {
					this.breadcrumbs[this.activeItemIndex + 2].append(view.$el);
				}
```

Reply:
Thank you for finding this!